templates/new/config/target.exs

```diff
@@ -23,19 +23,51 @@ if keys == [],
 config :nerves_firmware_ssh,
   authorized_keys: Enum.map(keys, &File.read!/1)
 
-# Configure nerves_init_gadget.
-# See https://hexdocs.pm/nerves_init_gadget/readme.html for more information.
+# Configure the network using vintage_net
+# See https://github.com/nerves-networking/vintage_net for more information
+config :vintage_net,
+  regulatory_domain: "US",
+  config: [
+    {"usb0", %{type: VintageNetDirect}},
+    {"eth0",
+      %{
+        type: VintageNetEthernet,
+        ipv4: %{ method: :dhcp }
+      }},
+    {"wlan0", %{ type: VintageNetWiFi }}
+  ]
+
+config :mdns_lite,
+  # The `host` key specifies what hostnames mdns_lite advertises.  `:hostname`
+  # advertises the device's hostname.local. For the official Nerves systems, this
+  # is "nerves-<4 digit serial#>.local".  mdns_lite also advertises
+  # "nerves.local" for convenience. If more than one Nerves device is on the
+  # network, delete "nerves" from the list.
 
-# Setting the node_name will enable Erlang Distribution.
-# Only enable this for prod if you understand the risks.
-node_name = if Mix.env() != :prod, do: "<%= app_name %>"
+  host: [:hostname, "nerves"],
+  ttl: 120,
```

Comment:
If it matters, the vintage_net and mdns_lite host and services have defaults configured at runtime in nerves_pack. So we don’t _need_ to have them as part of a config. There is prob value in having this all here, but I might vote to leave it out and instead put simple notes linking to docs to custom configure more if desired. Thoughts?

Reply:
My preference would be to put the vintage_net and mdns_lite config in the config.exs since it's more visible if you want to change it. I'm not sure we want runtime configuration in nerves_pack, since there's risk of it overwriting a user preference. For example, if a user purposefully removes an interface configuration to disable an interface, then it would be bad to reenable.

Reply:
Do we want `nerves_pack` to come in for `host` or just `@all_targets`? Putting application config entries in the config.exs that do not exist when running `mix test` or `mix run` will render warnings.

Reply:
Just `@all_targets`. I didn't mean to say that the config settings had to be in the `config.exs`. `target.exs` makes sense.